### PR TITLE
db genesis refactoring

### PR DESCRIPTION
**Pull request by alexplevako**

### CODE REVIEW

File: libraries/chain/include/scorum/chain/genesis.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    void init_witnesses();
+    void init_witness_schedule();
+    void init_global_property_object();
+    void init_rewards();
+
+private:
+    database& _db;
+    genesis_state_type _genesis_state;
+
+    bool _applied;
```
Comment (abolinbsu) [inline libraries/chain/include/scorum/chain/genesis.hpp]: Why do we need bool _applied; and void init();? Cann't we just do initialization in constructor?

File: libraries/chain/genesis.cpp [hunk 2]
```diff
@@ -90,23 +91,27 @@ void database::init_genesis(const genesis_state_type& genesis_state)
...
+    : _db(db)
+    , _genesis_state(genesis_state)
 {
-    const vector<genesis_state_type::account_type>& accounts = genesis_state.accounts;
+}
 
-    for (auto& account : accounts)
+void db_genesis::init_accounts()
+{
+    for (auto& account : _genesis_state.accounts)
```
Comment (Andrei-Masilevich) [inline libraries/chain/genesis.cpp]: Call dbs_account::create_initial_account here

File: libraries/chain/genesis.cpp [hunk 3]
```diff
@@ -94,9 +90,24 @@ void database::init_genesis(const genesis_state_type& genesis_state)
 scorum::chain::db_genesis::db_genesis(scorum::chain::database& db, const genesis_state_type& genesis_state)
     : _db(db)
     , _genesis_state(genesis_state)
+    , _applied(false)
```
Comment (Andrei-Masilevich) [inline libraries/chain/genesis.cpp]: _initialized will be better

File: libraries/chain/include/scorum/chain/genesis.hpp (hunk 1 above)
Comment (alexplevako) [reply libraries/chain/include/scorum/chain/genesis.hpp]: for unit tests. current implementation gives an ability inherit db_genesis in unit tests and test each init_* function

File: libraries/chain/genesis.cpp [hunk 4]
```diff
@@ -10,18 +10,18 @@
 
 #include <scorum/chain/pool/reward_pool.hpp>
 
+#include <scorum/chain/genesis.hpp>
+
 #include <fc/io/json.hpp>
 
-#define SCORUM_DEFAULT_INIT_PUBLIC_KEY "STM5omawYzkrPdcEEcFiwLdEu7a3znoJDSmerNgf96J2zaHZMTpWs"
+#define SCORUM_DEFAULT_INIT_PUBLIC_KEY "SCR5omawYzkrPdcEEcFiwLdEu7a3znoJDSmerNgf96J2zaHZMTpWs"
```
Comment (Andrei-Masilevich) [inline libraries/chain/genesis.cpp]: It is not necessary to create account "initdelegate" in generate_default_genesis_state 
(and public key without private key). Everyone need full json to run node in real world

**Pull request merged**